soc/nordic/common/dmm.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * nRF SoC specific public APIs for Device Memory Management (dmm) subsystem
+ */
+
+#ifndef SOC_NORDIC_COMMON_DMM_H_
+#define SOC_NORDIC_COMMON_DMM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * @brief Prepare a DMA output buffer for the specified device
+ *
+ * Allocate an output buffer in memory region that given device can perform DMA transfers from.
+ * Copy @p user_buffer contents into it.
+ * Writeback data cache lines associated with output buffer, if needed.
+ *
+ * @note Depending on provided user buffer parameters and SoC architecture,
+ *       dynamic allocation and cache operations might be skipped.
+ *
+ * @warning It is prohibited to read or write @p user_buffer or @p buffer_out contents
+ *          from the time this function is called until @ref dmm_dma_buffer_out_release()
+ *          is called on the same buffer or until this function returns with an error.
+ *
+ * @param dev Device to prepare the buffer for.
+ * @param user_buffer CPU address (virtual if applicable) of the buffer containing data
+ *                    to be processed by the given device.
+ * @param user_length Length of the buffer containing data to be processed by the given device.
+ * @param buffer_out Pointer to a bus address of a buffer containing the prepared DMA buffer.
+ *
+ * @retval 0 If succeeded.
+ * @retval -ENOMEM If output buffer could not be allocated.
+ * @retval -errno Negative errno for other failures.
+ */
+int dmm_dma_buffer_out_prepare(const struct device *dev, void const * user_buffer, size_t user_length, void ** buffer_out);
```

Comment:
general comment - now I see that this `_dma_` part could be removed from all of the APIs, as it feels to be redundant